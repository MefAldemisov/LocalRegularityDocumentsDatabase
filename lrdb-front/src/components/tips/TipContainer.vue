<style scoped>
aside {
    position: absolute;
    top: 0;
    right: -10rem;
    width: 15rem;
    z-index: 2;
}
</style>
<template>
    <aside>
        <the-load-tip v-if="tips.includes('load')"></the-load-tip>
        <the-nav-tip v-if="tips.includes('nav')"></the-nav-tip>
        <the-prev-tip v-if="tips.includes('prev')"></the-prev-tip>
        <the-submit-tip v-if="tips.includes('submit')"></the-submit-tip>
    </aside>
</template>
<script>
import TheLoadTip from "./templates/TheLoadTip.vue";
import TheNavTip from "./templates/TheNavTip.vue";
import ThePrevTip from "./templates/ThePrevTip.vue";
import TheSubmitTip from "./templates/TheSubmitTip.vue";
export default {
    name: "TipContainer",
    props: {
        tips: {
            type: Array,
            required: false,
        },
    },
    components: {
        TheLoadTip,
        TheNavTip,
        ThePrevTip,
        TheSubmitTip,
    },
};
</script>
