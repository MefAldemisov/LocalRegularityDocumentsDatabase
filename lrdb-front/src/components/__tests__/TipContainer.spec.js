import { shallowMount } from "@vue/test-utils";
import TipContainer from "../tips/TipContainer.vue";

describe("TipContainer", () => {
    const test_props = [
        [["nav"], true],
        [["load", "nav"], true],
        [["submit", "hi"], false],
        [["prev", "submit"], true],
        [["some", "vue"], false],
    ];

    it(`input is checked correctly`, () => {
        test_props.forEach((option) => {
            const tester = TipContainer.props;
            const res = tester.tips.validator(option[0]);
            expect(res).toBe(option[1]);
        });
    });

    it("should render to a snapshot when tips list has items", () => {
        const propsData = { tips: ["search", "load"] };
        const wrapper = shallowMount(TipContainer, { propsData });
        expect(wrapper.element).toMatchSnapshot();
    });

    it("the order of elements in the props shouldn't affect fie behaviour", () => {
        const propsData = { tips: ["load", "search"] };
        const wrapper = shallowMount(TipContainer, { propsData });
        expect(wrapper.element).toMatchSnapshot();
    });
});
