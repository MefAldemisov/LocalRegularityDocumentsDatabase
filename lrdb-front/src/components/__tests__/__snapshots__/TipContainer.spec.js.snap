// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TipContainer should render to a snapshot when tips list has items 1`] = `
<aside>
  <the-load-tip-stub />
   
  <!---->
   
  <!---->
   
  <!---->
</aside>
`;

exports[`TipContainer the order of elements in the props shouldn't affect fie behaviour 1`] = `
<aside>
  <the-load-tip-stub />
   
  <!---->
   
  <!---->
   
  <!---->
</aside>
`;
